core/src/tag_utils.rs

```diff
@@ -151,11 +151,59 @@ impl SwfMovie {
         Self::from_data(&data, url.into(), loader_url)
     }
 
-    /// Construct a movie based on the contents of the SWF datastream.
+    /// Construct a movie based on the contents of a datastream.
     pub fn from_data(
         swf_data: &[u8],
         url: String,
         loader_url: Option<String>,
+    ) -> Result<Self, Error> {
+        match Self::filter_from_projector_bundle(swf_data) {
+            Some(extracted_swf) => Self::from_swf_data(&extracted_swf, url, loader_url),
+            None => Self::from_swf_data(swf_data, url, loader_url),
+        }
+    }
+
+    /// Get an SWF from data if the data is a swf/projector bundle
+    fn filter_from_projector_bundle(data: &[u8]) -> Option<Vec<u8>> {
+        let len = data.len();
+        if len < 8 {
+            return None;
+        }
+
+        let signature = u32::from_le_bytes(
+            data[len - 8..len - 4]
+                .try_into()
+                .expect("4 bytes make a u32"),
+        );
+
+        if signature != 0xFA123456 {
+            return None;
+        }
```

Comment:
Suggestion, just for simpler code: `data[len - 8..len - 4] != [0xFA, 0x12, 0x34, 0x56]`. No need to convert to an `u32` necessarily.
Again, this is just a minor nit, not a blocker for merging.

Reply:
Yep, I originally did it this way because I liked the "readable" way the u32 flowed "..123456". The array would have that reversed. But you're right that it'd be a smaller line of code so that's a fair point.